src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -969,10 +969,25 @@ because xbuild doesn't support framework reference assemblies.
 />
 
 <Target Name="UpdateAndroidResources"
-    Condition=" '$(DesignTimeBuild)' != 'True' Or '$(DeferredBuildSupported)' != 'True' Or '$(DeferredBuild)' == 'True' "
+    Condition=" ('$(DesignTimeBuild)' != 'True' Or '$(DeferredBuildSupported)' != 'True' Or '$(DeferredBuild)' == 'True') And !Exists ('$(_AndroidStampDirectory)_UpdateAndroidResources.stamp')"
     DependsOnTargets="_SetupDesignTimeBuildForCompile;_ManagedUpdateAndroidResgen;_UpdateAndroidResources"
 />
 
+<Target Name="_BeforeUpdateAndroidResources"
+    BeforeTargets="_SetupDesignTimeBuildForCompile"
```

Comment:
note we use before and after targets in these new ones because we do want these to run even if there is an error.